extensions/flac/src/main/jni/flac_parser.cc

```diff
@@ -174,24 +174,16 @@ void FLACParser::metadataCallback(const FLAC__StreamMetadata *metadata) {
       break;
     case FLAC__METADATA_TYPE_VORBIS_COMMENT:
       if (!mVorbisCommentValid) {
-        FLAC__uint32 count = 0;
-        const FLAC__StreamMetadata_VorbisComment *vc =
-            &metadata->data.vorbis_comment;
-        mVorbisCommentValid = true;
-        mVorbisComment.metadataArray =
-            (char **) malloc(vc->num_comments * sizeof(char *));
-        for (FLAC__uint32 i = 0; i < vc->num_comments; ++i) {
-          FLAC__StreamMetadata_VorbisComment_Entry *vce = &vc->comments[i];
-          if (vce->entry != NULL) {
-            mVorbisComment.metadataArray[count] =
-                (char *) malloc((vce->length + 1) * sizeof(char));
-            memcpy(mVorbisComment.metadataArray[count], vce->entry,
-                vce->length);
-            mVorbisComment.metadataArray[count][vce->length] = '\0';
-            count++;
+        FLAC__StreamMetadata_VorbisComment vc = metadata->data.vorbis_comment;
+        for (FLAC__uint32 i = 0; i < vc.num_comments; ++i) {
+          FLAC__StreamMetadata_VorbisComment_Entry vce = vc.comments[i];
+          if (vce.entry != NULL) {
+            std::string comment(reinterpret_cast<char *>(vce.entry),
```

Comment:
Removed the mallocs here and the free. Using `std::vector` and `std::string` instead.